parse_config/parse_config.cpp

```diff
@@ -0,0 +1,336 @@
+			}
+			Block	newBlock("http");
+			localBlock.blocks.push_back(newBlock);
+			it = fillBlocks(it + 2, depth + 1, *(localBlock.blocks.end() - 1));
+		} else if (*it == "server") {
+			if (depth != 1) {
+				throw "Error: in config file 'server' block's position is wrong";
+			}
+			Block	newBlock("server");
+			localBlock.blocks.push_back(newBlock);
+			it = fillBlocks(it + 2, depth + 1, *(localBlock.blocks.end() - 1));
+		} else if (*it == "location") {
+			if (depth != 2) {
+				throw "Error: in config file 'location' block's position is wrong";
+			}
+			Block	newBlock("location");
+			it = setLocation(it, newBlock);
+			localBlock.blocks.push_back(newBlock);
+			it = fillBlocks(it + 1, depth + 1, *(localBlock.blocks.end() - 1));
+		} else if (*it == "}") {
+			return (it);
+		} else if (isSkipBlock(it) == true) {
+			it = skipThisBlock(it);
+		} else if (*it == ";") {
+			continue ;
+		}
+		else {
+			string	key = *it;
+			it++;
+			localBlock.directives[key] = Block::getValue(it);
+		}
+	}
+	return (it);
+}
+
+/*fillBlocks 함수에서 블록을 스킵할지 정하는 함수*/
+bool	ParseConfig::isSkipBlock(vector<string>::iterator it)
+{
+	while (1)
+	{
+		if (*it == "{") {
+			return (true);
+		} else if (*it == ";") {
+			return (false);
+		}
+		it++;
+	}
+}
+
+/*블록을 스킵하는 함수*/
+vector<string>::iterator	ParseConfig::skipThisBlock(vector<string>::iterator it) {
+	while (*it != "}") {
+		++it;
+	}
+	return (it);
+}
+
+/*Location 의 주소를 저장하고 { 위치의 iterator를 return*/
+vector<string>::iterator	ParseConfig::setLocation(vector<string>::iterator it, Block& block) {
+	while (*it != "{") {
+		++it;
+	}
+	block.location = *(it - 1);
+	return (it);
+}
+
+/* charset이 섞인 문자덩어리를 charset과 문자열로 분리해서 words 벡터에 pushback 하는 함수 */
+void	ParseConfig::pushMixedWords(const string& word, vector<string>& words) {
+	for (size_t i = 0; i < word.size(); i++) {
+		string	new_word;
+		words.push_back(new_word);
+		if (isCharInCharset(word[i], charset) == true) {
+			(words.end() - 1)->push_back(word[i]);
+		} else {
+			for (; isCharInCharset(word[i], charset) == false && word[i] != 0; i++) {
+				(words.end() - 1)->push_back(word[i]);
+			}
+			if (isCharInCharset(word[i], charset) == true) {
+				i--;
+			}
+		}
+	}
+}
+/* string에서 다음 문자 덩어리를 찾아서 return 하는 함수 */
+string	ParseConfig::nextWord(string line, size_t& idx) {
+	string	new_word;
+
+	while (isspace(line[idx]) == true) {
+		idx++;
+	}
+	for (; (idx < line.size()) && isspace(line[idx]) == false; ++idx) {
+		new_word.push_back(line[idx]);
+	}
+	return (new_word);
+}
+/* string 에 charset 의 문자가 포함되는지 확인하는 함수 */
+bool	ParseConfig::isCharsetInString(const string& str, const string charset) {
+	for (size_t i = 0; i < str.size(); i++) {
+		if (isCharInCharset(str[i], charset) == true) {
+			return (true);
+		}
+	}
+	return (false);
+}
+/* char c가 charset 에 포함되는지 확인하는 함수 */
+bool	ParseConfig::isCharInCharset(char c, const string charset) {
+	for (size_t i = 0; i < charset.size(); i++) {
+		if (c == charset[i]) {
+			return (true);
+		}
+	}
```

Comment:
```suggestion
	return find(charset.begin(), charset.end(), c) != charset.end() ? true : false;
```
같은 방식으로도 가능합니다! 근데 성능은 큰차이가 없긴해서 추후에 코드를 줄이고 싶거나 그러시면 참고하시면 좋을거 같아요!

Reply:
오 꿀팁 감사합니다.